lib/graphql/execution/lazy.rb

```diff
@@ -20,18 +20,39 @@ def self.resolve(val)
       end
 
       # Create a {Lazy} which will get its inner value by calling the block
-      # @param get_value_func [Proc] a block to get the inner value (later)
-      def initialize(&get_value_func)
-        @get_value_func = get_value_func
+      # @param value_proc [Proc] a block to get the inner value (later)
+      def initialize(original = nil, value:, exec:)
```

Comment:
Lazy objects now have two blocks defined on them `value` is the previous `&get_value_func` which resolves the object. `exec` is the new block which is the step that can issue a concurrent execution.